src/anomalib/data/base/video.py

```diff
@@ -148,22 +149,14 @@ def __getitem__(self, index: int) -> dict[str, str | torch.Tensor]:
         item["original_image"] = item["image"].to(torch.uint8)
 
         # apply transforms
-        if "mask" in item and item["mask"] is not None:
-            processed_frames = [
-                self.transform(image=frame.numpy(), mask=mask)
-                for frame, mask in zip(item["image"], item["mask"], strict=True)
-            ]
-            item["image"] = torch.stack([item["image"] for item in processed_frames]).squeeze(0)
-            mask = torch.as_tensor(item["mask"])
-            item["mask"] = torch.stack([item["mask"] for item in processed_frames]).squeeze(0)
-            item["label"] = torch.Tensor([1 in frame for frame in mask]).int().squeeze(0)
+        if item.get("mask") is not None:
+            item["image"], item["mask"] = self.transform(item["image"], Mask(item["mask"]))
+            item["label"] = torch.Tensor([1 in frame for frame in item["mask"]]).int().squeeze(0)
```

Comment:
Maybe not related to this PR but should we consider using torch.book for labels across the repo? We can save on some gpu memory.